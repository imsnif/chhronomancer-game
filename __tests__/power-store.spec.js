beforeEach(() => {
  jest.resetModules()
})

test('PowerStore => is constructed properly', () => {
  const powerStore = require('../stores/power').default
  const powers = powerStore.powers.map(p => p)
  expect(powers).toEqual([])
})

test('PowerStore => can add a power', () => {
  const powerStore = require('../stores/power').default
  powerStore.addPower({
    playerId: 1,
    timelineName: 'foo',
    name: 'bar',
    startTime: 1,
    endTime: 2,
    target: 'baz',
    allies: [{a: 1, b: 1}, {c: 3, d: 4}],
    enemies: [{x: 1, y: 1}, {z: 3, q: 4}]
  })
  const powers = powerStore.powers.map(p => p)
  expect(powers).toMatchSnapshot()
})

test('PowerStore => can get power', () => {
  const powerStore = require('../stores/power').default
  powerStore.addPower({
    playerId: 1,
    timelineName: 'foo'
  })
  expect(powerStore.getPower(1, 'foo')).toMatchSnapshot()
})

test('PowerStore => can get progress', () => {
  const powerStore = require('../stores/power').default
  powerStore.addPower({
    playerId: 1,
    timelineName: 'foo',
    startTime: 1000,
    endTime: 2000
  })
  expect(powerStore.getProgress(1, 'foo', 1500)).toEqual(50)
})

test('PowerStore => can get timeLeft', () => {
  const powerStore = require('../stores/power').default
  powerStore.addPower({
    playerId: 1,
    timelineName: 'foo',
    startTime: 10000,
    endTime: 20000
  })
  expect(powerStore.getTimeLeft(1, 'foo', 15000)).toMatchSnapshot()
})
