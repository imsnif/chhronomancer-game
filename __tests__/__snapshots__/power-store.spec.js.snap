// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerStore => can add a power 1`] = `
Array [
  Object {
    "allies": Array [
      Object {
        "a": 1,
        "b": 1,
      },
      Object {
        "c": 3,
        "d": 4,
      },
    ],
    "endTime": 2,
    "enemies": Array [
      Object {
        "x": 1,
        "y": 1,
      },
      Object {
        "q": 4,
        "z": 3,
      },
    ],
    "name": "bar",
    "playerId": 1,
    "startTime": 1,
    "target": "baz",
    "timelineName": "foo",
  },
]
`;

exports[`PowerStore => can get power 1`] = `
Object {
  "allies": undefined,
  "endTime": undefined,
  "enemies": undefined,
  "name": undefined,
  "playerId": 1,
  "startTime": undefined,
  "target": undefined,
  "timelineName": "foo",
}
`;

exports[`PowerStore => can get timeLeft 1`] = `"00:00:00"`;
